pandas/_libs/tslibs/period.pyx

```diff
@@ -2404,6 +2405,14 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
+            try:
+                ts = Timestamp(value)
+            except ValueError:
```

Comment:
we end up parsing this twice which is not good, yeah i think this needs to be integrated a bit more to `parse_time_string`.

i guess the currently soln would be ok in the interim, but would need to run asv's on all periods to see what kind of perf hit here.

Reply:
As far I read, nanosecond was not accepted into dateutil module. This interim solution could stand for a while.
I stay tuned...

Reply:
> As far I read, nanosecond was not accepted into dateutil module. This interim solution could stand for a while.
> I stay tuned...

we don't wait for dateutil in this, as it will not likey every support nanosecond because the standard library does not